### PR TITLE
Remove non-source-generator sorter, restructure NuGet package

### PR DESCRIPTION
**Pull request by jonathanpeppers**
## Summary

Removes the old `NetworkSort` static class and all auto-generated sorting code (~4MB), making the Roslyn source generator the only way to use sorting networks. The `SortingNetworks` NuGet package is restructured from a library into a thin attribute + bundled source generator package.

### Deleted (~48K lines removed)
- `SortingNetworks/NetworkSort.cs`, `NetworkSort.Unrolled.cs`, `NetworkSort.Simd.cs`, `NetworkSort.Simd.Arm.cs`, `NetworkData.cs` - the old auto-generated sorting code
- `scripts/generate_unrolled.cs` - the code generation script
- `SortingNetworks.Tests/NetworkSortTests.cs` - old API tests (unique tests migrated)
- `SortingNetworks.Benchmarks/StringSortingBenchmarks.cs` - pending string support in generator

### NuGet package restructured
The `SortingNetworks` NuGet package now contains:
- `lib/netstandard2.0/SortingNetworks.dll` - just the `SortingNetworkAttribute` class
- `analyzers/dotnet/cs/SortingNetworks.Generators.dll` - the Roslyn source generator

### New consumer API
```csharp
using SortingNetworks;

[SortingNetwork(27, typeof(int))]
[SortingNetwork(28, typeof(int))]
partial class MySorter { }

MySorter.Sort(array);
MySorter.Sort(span);
MySorter.Sort(span, comparer);
```

### Tests (232 passing)
- Migrated all unique test patterns from `NetworkSortTests` to `GeneratedSortTests`
- Stress tests (100 seeds) for all 13 primitive types at sizes 27/28
- Edge cases: already sorted, reversed, duplicates, min/max extremes
- Custom comparer tests, null handling, wrong-size throws

### Benchmarks
- Removed `NetworkSort`/`NetworkSort_Span` from all benchmark files
- Added `GeneratedSort` to `NIntSortingBenchmarks` and `NUIntSortingBenchmarks`
- Added nint/nuint sizes 23-32 to benchmark `GeneratedSorters.cs`

### Documentation
- README rewritten for source-generator-only usage

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -278,21 +277,20 @@ sorting network dominates:
...
 #### string (specialized `Sort(string[])` path)
 
-The specialized `Sort(string[])` overload uses `string.CompareOrdinal` in the
-unrolled network, avoiding `IComparer<T>` interface dispatch overhead:
+The generated `Sort(Span<string>)` method uses `string.CompareOrdinal` in the unrolled network, avoiding `IComparer<T>` interface dispatch overhead:
 
-| Type | ArraySort (27) | NetworkSort (27) | Speedup |
+| Type | ArraySort (27) | GeneratedSort (27) | Speedup |
 |---|---|---|---|
 | string | 979 ns | 527 ns | **1.9x** |
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply README.md]: This is now valid -- string support was added to the generator (merged from main, which includes PR #89). The generator emits string.CompareOrdinal-based scalar sort for string types. I updated the section heading from 'specialized Sort(string[]) path' to 'scalar string.CompareOrdinal path' to be more accurate.

File: README.md [hunk 2]
```diff
@@ -2,53 +2,50 @@
...
-Overloads are provided for every primitive .NET type: `byte`, `sbyte`,
+Overloads are generated for every primitive .NET type: `byte`, `sbyte`,
 `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `nint`, `nuint`,
 `char`, `float`, and `double`.
 
 ## Usage
 
+Add the `SortingNetworks` NuGet package, then decorate a `partial class` with
+`[SortingNetwork(size, typeof(T))]` for each size/type combination you need:
+
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply README.md:24]: Good catch. Added a note to the Usage section clarifying that the generated SIMD code requires .NET 8+ (System.Runtime.Intrinsics) and the scalar path requires .NET 5+ (Unsafe). The analyzer itself can't enforce a minimum TFM since it runs in the compiler -- but the generated code will produce compile errors on older targets. A future improvement could detect the TFM and conditionally skip SIMD emission.
